src/signals/my-incidents/context/provider.tsx

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2022 Gemeente Amsterdam
+import type { FC } from 'react'
+import { useMemo } from 'react'
+
+import type { MyIncidentsValue } from '../types'
+import { MyIncidentsContext } from './context'
+
+export const initialValue: MyIncidentsValue = {
+  email: undefined,
+  setEmail: () => {},
+}
+
+interface Props {
+  value?: MyIncidentsValue
+}
+
+export const MyIncidentsProvider: FC<Props> = ({ value, children }) => {
+  const newValue = useMemo(
+    () => ({
+      ...initialValue,
+      ...value,
+    }),
+    [value]
+  )
+
+  return (
+    <MyIncidentsContext.Provider value={newValue}>
+      {children}
```

Comment:
to make this future proof , consider adding children into props

https://stackoverflow.com/questions/71788254/react-18-typescript-children-fc